connect/runtime/src/main/java/org/apache/kafka/connect/runtime/ExactlyOnceWorkerSourceTask.java

```diff
@@ -256,14 +257,25 @@ private void maybeBeginTransaction() {
     private void commitTransaction() {
         log.debug("{} Committing offsets", this);
 
+        long commitTimeoutMs = workerConfig.getLong(WorkerConfig.OFFSET_COMMIT_TIMEOUT_MS_CONFIG);
         long started = time.milliseconds();
+        long deadline = started + commitTimeoutMs;
 
         // We might have just aborted a transaction, in which case we'll have to begin a new one
         // in order to commit offsets
         maybeBeginTransaction();
 
         AtomicReference<Throwable> flushError = new AtomicReference<>();
-        if (offsetWriter.beginFlush()) {
+        boolean shouldFlush = false;
+        try {
+            // Provide a constant timeout value to wait indefinitely, as there should not be any concurrent flushes.
+            // This is because commitTransaction is always called on the same thread, and should always block until
+            // the flush is complete, or cancel the flush if an error occurs.
+            shouldFlush = offsetWriter.beginFlush(deadline - time.milliseconds(), TimeUnit.MILLISECONDS);
```

Comment:
I agree with the comment about how this method should never be invoked while there are in-progress flushes.

Given that, is there any reason to go to the work of calculating a deadline and deriving a timeout from it, instead of simply invoking this method with a timeout of zero?

We could even add a no-arg variant of `beginFlush` that calls `beginFlush(0, TimeUnit.MILLISECONDS)`.